ansible-playbooks/roles/deploy-acm-stack/tasks/main.yml

```diff
@@ -29,154 +29,163 @@
     - AWS_SECRET_KEY
     - SLACK_API_URL
 
-- name: Deploy Openshift GitOps
-  shell: |-
-    if [ {{ ENV }} != "local" ]; then
-      sed -i'' --expression 's@<VAULT_ADDRESS>@'{{ VAULT_ADDRESS }}'@g;s@<VAULT_TOKEN>@'{{ VAULT_TOKEN }}'@g' ./../../cluster-bootstrap/openshift-gitops/config/argocd.yaml
-    fi
-
-    printf "=====================Create Openshift Gitops Subscription...\n"
-    kubectl apply -k ./../../cluster-bootstrap/openshift-gitops/deploy
-
-    while [[ $(kubectl get ns openshift-gitops --no-headers --ignore-not-found | wc -l) -eq 0 ]]; 
-        do echo "Waiting for openshift-gitops namespace to be created" && sleep 30;
-    done
+- name: Update argocd configuration
+  template:
+    src: argocd.yaml.j2
+    dest: ./../../cluster-bootstrap/openshift-gitops/config/argocd.yaml
 
-    printf "=====================Waiting for Openshift Gitops start up and running...\n"
-    kubectl wait --for=condition=Ready pods --all -n openshift-gitops --timeout=5m
-    printf "=====================Openshift Gitops deploy successful!\n"
-
-    printf "=====================Create and config Openshift Gitops instance with Vault plugin configed...\n"
-    kubectl apply -k ./../../cluster-bootstrap/openshift-gitops/config
-
-    printf "=====================Waiting for configured Gitops repo server up and running...\n"
-    kubectl wait --for=condition=Ready pods --all -n openshift-gitops --timeout=5m
-    printf "=====================Openshift Gitops deploy successful!\n"
-    sleep 10
+- name: Apply openshift-gitops subscription
+  kubernetes.core.k8s:
+    state: present
+    src: "{{ item }}"
+  loop:
+    - "{{ playbook_dir }}/../../cluster-bootstrap/openshift-gitops/deploy/subscription.yaml"
+
+- name: Wait namespace openshift-gitops is created
+  kubernetes.core.k8s_info:
+    kind: Namespace
+    wait: yes
+    name: openshift-gitops
+    namespace: openshift-gitops
+    wait_sleep: 5
+    wait_timeout: 30
+
+- name: Wait all pods to be ready
+  kubernetes.core.k8s_info:
+    kind: Pod
+    wait: yes
+    wait_condition:
+      type: Ready
+      status: "True"
+    namespace: openshift-gitops
+    wait_sleep: 10
+    wait_timeout: 300
+
+- name: Apply the openshift-gitops configuration
+  kubernetes.core.k8s:
+    state: present
+    src: "{{ item }}"
+  loop:
+    - "{{ playbook_dir }}/../../cluster-bootstrap/openshift-gitops/config/argocd.yaml"
+    - "{{ playbook_dir }}/../../cluster-bootstrap/openshift-gitops/config/repository.yaml"
+    - "{{ playbook_dir }}/../../cluster-bootstrap/openshift-gitops/config/rolebinding.yaml"
 
+- name: Wait for Openshift GitOps
+  shell: |-
     while [[ $(kubectl get pods -n openshift-gitops --no-headers --ignore-not-found | awk '/openshift-gitops-repo-server/' | wc -l) -ne 1 ]]; 
```

Comment:
could you also replace also this with a wait like you did for namespaces? sorry for missing this yesterday

Reply:
Updated. thx